tools/uart_hw_test/README.md

```diff
@@ -0,0 +1,54 @@
+# UART HW test tool
```

Comment:
This tool is only valid with uart_hw_test app. How about moving this folder under apps/examples/uart_hw_test folder?

Reply:
I dont think its a good idea to move the tool to src folder. It will be difficult to search and run the tool. 
But, I have added a dependency in the script used to run the tool. It will run only if the corresponding config is enabled.